tools/jest/helpers/mock-xhr.js

```diff
@@ -0,0 +1,20 @@
+export class MockXhr {
+  static originalXHR = null
+
+  static setup () {
+    MockXhr.originalXHR = window.XMLHttpRequest
+    window.XMLHttpRequest = jest.fn(() => new MockXhr())
+  }
+
+  static teardown () {
+    if (MockXhr.originalXHR) {
+      window.XMLHttpRequest = MockXhr.originalXHR
+      MockXhr.originalXHR = null
+    }
+  }
+
+  open = jest.fn()
+  send = jest.fn()
```

Comment:
Probably wouldn't want to use fake-timer here or we'd affect every file(?)
```suggestion
  onload = jest.fn()
  send = jest.fn(() => {
    setTimeout(this.onload, Math.floor(Math.random()*10))
  })
```